pkg/cloud/httpsink/http_storage.go

```diff
@@ -266,7 +266,11 @@ func (h *httpStorage) req(
 		_ = resp.Body.Close()
 		err := errors.Errorf("error response from server: %s %q", resp.Status, body)
 		if err != nil && resp.StatusCode == 404 {
-			err = errors.Wrapf(cloud.ErrFileDoesNotExist, "http storage file does not exist: %s", err.Error())
+			err = errors.WithDetailf(
```

Comment:
Hm. So here/elsewhere in the cloud errors, I'm wondering if we want to move this to the hint instead of keeping it in the main error text? If my recollection is correct, some/most client libraries only default to the main error text, so if we change this, do we risk having support requests where a user comes back to us with "hey, our backup airflow job failed with 'pq: object does not exist'" without the actual error we got from the request?

Reply:
hm, yeah, if the other error's text is considered critical info in order to be able to debug the problem, then it makes more sense to put it in the main text.